File: practice/src/main/java/com/decade/practice/ProductionApplication.java
package com.decade.practice;

import com.decade.practice.database.repository.UserRepository;
import com.decade.practice.entities.domain.embeddable.ImageSpec;
import com.decade.practice.entities.domain.entity.Chat;
import com.decade.practice.entities.domain.entity.TextEvent;
import com.decade.practice.usecases.core.EventStore;
import com.decade.practice.usecases.core.OnlineStatistic;
import com.decade.practice.usecases.core.UserOperations;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ApplicationContext;
import org.springframework.transaction.PlatformTransactionManager;
import org.springframework.transaction.TransactionDefinition;
import org.springframework.transaction.support.TransactionTemplate;

import java.time.Instant;
import java.util.Date;

@SpringBootApplication
public class ProductionApplication {

      private static final String MALE = "male";

      public static void main(String[] args) {
            SpringApplication app = createApp(ProductionApplication.class);
            ApplicationContext context = app.run(args);
            initialize(context);
      }

      private static boolean isRunningInContainer() {
            return System.getenv().containsKey("DECADE");
      }

      public static SpringApplication createApp(Class<?> appClass) {
            SpringApplication app = new SpringApplication(appClass);
            System.out.println("isRunningInContainer:" + isRunningInContainer());
            System.out.println("environments: " + System.getenv().toString() + "\n\n\n");
            if (isRunningInContainer()) {
                  app.setAdditionalProfiles("docker");
            }
            return app;
      }

      public static void initialize(ApplicationContext context) {
            UserOperations userService = context.getBean(UserOperations.class);
            UserRepository userRepo = context.getBean(UserRepository.class);
            PlatformTransactionManager txManager = context.getBean(PlatformTransactionManager.class);
            TransactionTemplate transactionOperations = new TransactionTemplate(txManager);
            transactionOperations.setIsolationLevel(TransactionDefinition.ISOLATION_READ_COMMITTED);

            EventStore eventStore = context.getBean(EventStore.class);
            OnlineStatistic onlineStat = context.getBean(OnlineStatistic.class);

            userService.create(
                  "Luffy",
                  "Luffy",
                  "Luffy",
                  new Date(),
                  MALE,
                  new ImageSpec("http://192.168.3.104:8080/image?filename=luffy.jpeg", "luffy.jpeg", 512, 512, "jpg"),
                  true
            );
            userService.create(
                  "Nami",
                  "Nami",
                  "Nami",
                  new Date(),
                  MALE,
                  new ImageSpec("http://192.168.3.104:8080/image?filename=nami.jpeg", "nami.jpeg", 512, 512, "jpg"),
                  true
            );
            userService.create(
                  "Chopper",
                  "Chopper",
                  "Chopper",
                  new Date(),
                  MALE,
                  new ImageSpec("http://192.168.3.104:8080/image?filename=chopper.jpeg", "chopper.jpeg", 512, 512, "jpg"),
                  true
            );

            transactionOperations.executeWithoutResult(status -> {
                  var luffy = userRepo.getByUsername("Luffy");
                  var nami = userRepo.getByUsername("Nami");
                  var chopper = userRepo.getByUsername("Chopper");

                  TextEvent event1 = new TextEvent(new Chat(luffy, nami), nami, "Hello");
                  event1.setCreatedTime(System.currentTimeMillis() - 5 * 60 * 1000);
                  eventStore.save(event1);

                  TextEvent event2 = new TextEvent(new Chat(luffy, chopper), chopper, "Ekk");
                  event2.setCreatedTime(System.currentTimeMillis() - 10 * 60 * 1000);
                  eventStore.save(event2);

                  TextEvent event3 = new TextEvent(new Chat(luffy, chopper), chopper, "Vcl");
                  event3.setCreatedTime(System.currentTimeMillis() - 5 * 60 * 1000);
                  eventStore.save(event3);

                  onlineStat.set(nami, Instant.now().getEpochSecond() - 2 * 60);
                  onlineStat.set(chopper, Instant.now().getEpochSecond() - 10 * 60);
            });
      }
}

File: practice/src/main/java/com/decade/practice/ProductionApplication.kt
package com.decade.practice

import com.decade.practice.core.EventStore
import com.decade.practice.core.OnlineStatistic
import com.decade.practice.core.UserOperations
import com.decade.practice.database.repository.UserRepository
import com.decade.practice.model.domain.embeddable.ImageSpec
import com.decade.practice.model.domain.entity.Chat
import com.decade.practice.model.domain.entity.MALE
import com.decade.practice.model.domain.entity.TextEvent
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.ApplicationContext
import org.springframework.transaction.PlatformTransactionManager
import org.springframework.transaction.TransactionDefinition
import org.springframework.transaction.support.TransactionTemplate
import java.time.Instant
import java.util.*


@SpringBootApplication
class ProductionApplication

fun main(args: Array<String>) {
      val app = createApp(ProductionApplication::class.java)
      val context = app.run(*args)
      initialize(context)
}

private fun isRunningInContainer() =
      System.getenv().containsKey("DECADE")

fun createApp(appClass: Class<*>): SpringApplication {
      val app = SpringApplication(appClass)
      println("isRunningInContainer:" + isRunningInContainer())
      println("environments: " + System.getenv().toString() + "\n\n\n")
      if (isRunningInContainer()) {
            app.setAdditionalProfiles("docker")
      }
      return app
}

fun initialize(context: ApplicationContext) {
      val userService = context.getBean(UserOperations::class.java)
      val userRepo = context.getBean(UserRepository::class.java)
      val txManager = context.getBean(PlatformTransactionManager::class.java)
      val transactionOperations = TransactionTemplate(txManager).apply {
            isolationLevel = TransactionDefinition.ISOLATION_READ_COMMITTED
      }
      val eventStore = context.getBean(EventStore::class.java)
      val onlineStat = context.getBean(OnlineStatistic::class.java)

      userService.create(
            "Luffy",
            "Luffy",
            "Luffy",
            Date(),
            MALE,
            ImageSpec("http://192.168.3.104:8080/image?filename=luffy.jpeg", "luffy.jpeg"),
            true
      )
      userService.create(
            "Nami",
            "Nami",
            "Nami",
            Date(),
            MALE,
            ImageSpec("http://192.168.3.104:8080/image?filename=nami.jpeg", "nami.jpeg"),
            true
      )
      userService.create(
            "Chopper",
            "Chopper",
            "Chopper",
            Date(),
            MALE,
            ImageSpec("http://192.168.3.104:8080/image?filename=chopper.jpeg", "chopper.jpeg"),
            true
      )

      transactionOperations.executeWithoutResult {

            val luffy = userRepo.getByUsername("Luffy")
            val nami = userRepo.getByUsername("Nami")
            val chopper = userRepo.getByUsername("Chopper")
            eventStore.save(event = TextEvent(Chat(luffy, nami), nami, "Hello").apply {
                  createdTime = System.currentTimeMillis() - 5 * 60 * 1000
            })
            eventStore.save(event = TextEvent(Chat(luffy, chopper), chopper, "Ekk").apply {
                  createdTime = System.currentTimeMillis() - 10 * 60 * 1000
            })
            eventStore.save(event = TextEvent(Chat(luffy, chopper), chopper, "Vcl").apply {
                  createdTime = System.currentTimeMillis() - 5 * 60 * 1000
            })

            onlineStat.set(nami, Instant.now().epochSecond - 2 * 60)
            onlineStat.set(chopper, Instant.now().epochSecond - 10 * 60)
      }
}